packages/core/src/types.ts

```diff
@@ -217,6 +217,15 @@ export type Histogram = BasePlot & {
 
 export type Plot = Scatterplot | Histogram;
 
+// eslint-disable-next-line no-shadow
+export enum AttributePlotType {
+  BoxPlot = 'Box Plot',
+  DotPlot = 'Dot Plot',
+  StripPlot = 'Strip Plot',
+}
+
+export type AttributePlots = Record<string, `${AttributePlotType}`>;
```

Comment:
Document type. It wasn't clear to me until reading the rest of the code why this needs to be a `Record`, ie, what does the `string` represent?

Reply:
Record here is simply a more concise way of doing this notation:

```js
type AttributePlots = {
  [attribute: string]: keyof typeof AttributePlotType
}
```

Enums don't behave super well in typescript as opposed to languages like Java, IMO. I'll note that in the documentation. I think moving forward, we should try to use Record as it is more clear and easier to read.

Reply:
Sorry, my original comment wasn't clear: I think you should note that the `string` is the name of the attribute. So this record maps attribute names to their plot types.